greedy_max/__init__.py

```diff
@@ -532,39 +675,77 @@ def schedule(self):
         sum_score = 0.0
         time_used = 0
         n_iter = 0
+        all_obs = get_observations(self.observations)
 
         # Unscheduled time slots.
         while self.plan.timeslots_not_scheduled() != 0:
 
-            plan, obstab, targtab = self._run()
+            self._run()
             n_iter += 1
 
             # Fill nightly plan one observation at a time.
             sum_score = 0.0
             time_used = 0
 
             # Print current plan
-            # TODO: Move this to inside of plan. as a summary method
-            print('Iteration {:4d}'.format(n_iter))
+            print(f'Iteration {n_iter:4d}')
             for site in self.sites:
                 print(Site(site).name.upper())
-                print('{:18} {:>9} {:>8} {:>8} {:>8}'.format('Obsid', 'obs_order', 'i_start', 'i_end', 'Max W'))
-                obs_order, i_start, i_end = get_order(plan=plan.schedule[site])
-
-                for i in range(len(obs_order)):
-                    if obs_order[i] >= 0:
-                        print('{:18} {:>9d} {:>8d} {:>8d} {:8.4f}'.format(short_observation_id(self.observations[obs_order[i]].name),
-                                                                          obs_order[i], i_start[i], i_end[i],
-                                                                          np.max(abs(
-                                                                              self.time_slots.weights[site][
-                                                                                  obs_order[i]][
-                                                                                  i_start[i]:i_end[i] + 1]))))
-                        sum_score += np.sum(abs(self.time_slots.weights[site][obs_order[i]][i_start[i]:i_end[i] + 1]))
-                        time_used += (i_end[i] - i_start[i] + 1)
-            # input()
-
-        print('Sum score = {:7.2f}'.format(sum_score))
-        print('Sum score/time step = {:7.2f}'.format(sum_score / (2 * self.time_slots.total)))
-        print('Time scheduled = {:5.2f}'.format(time_used * self.time_slots.slot_length.to(u.hr)))
+                obs_in_plan = self.plan.get_observation_orders(site)
+
+                output_table = []
+
+                for observation in obs_in_plan:
+                    obs_idx = observation[0]
+                    start = observation[1]
+                    end = observation[2]   
+                    if obs_idx != EMPTY and obs_idx != UNSCHEDULABLE:
+                        unit = self.plan.get_unit_by_observation(site, obs_idx) # TODO: handle None case
```

Comment:
If `unit` is none it will crash the program. But that would mean that Gmax schedule an observation outside a unit... which is weird if not impossible, as we don't have a way to change the composition of the unit in runtime.